File: computer_science/data_structures/linked_list/linked_list.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None


class LinkedList:
    def __init__(self):
        self.head = None

    def is_empty(self):
        return self.head == None

    def push_front(self, data):
        new_node = Node(data)
        new_node.next = self.head
        self.head = new_node

    def push_back(self, data):
        new_node = Node(data)
        current_node = self.head

        while current_node.next:
            current_node = current_node.next

        current_node.next = new_node

    def size(self):
        current_node = self.head
        count = 0

        while current_node:
            count += 1
            current_node = current_node.next

        return count

    def search(self, item):
        current_node = self.head

        while current_node and not current_node.data == item:
            current_node = current_node.next

        return current_node != None and current_node.data == item

    def remove(self, item):
        if self.head == None:
            print('List is empty')
            return

        current_node = self.head
        previous_node = None

        while current_node and current_node.data != item:
            previous_node = current_node
            current_node = current_node.next

        if previous_node == None:
            self.head = current_node.next
        else:
            previous_node.next = current_node.next

File: computer_science/data_structures/linked_list/linked-list.js
export class Node {
  constructor(value, next = null) {
    this.value = value;
    this.next = next;
  }
}

export class LinkedList {
  constructor() {
    this.head = null;
  }

  pushFront(value) {
    this.head = new Node(value, this.head);
  }

  pushBack(value) {
    if (this.isEmpty()) {
      this.head = new Node(value);
      return;
    }

    let currentNode = this.head;

    while (currentNode && currentNode.next) {
      currentNode = currentNode.next;
    }

    currentNode.next = new Node(value);
  }

  size() {
    let count = 0;
    let currentNode = this.head;

    while (currentNode) {
      count += 1;
      currentNode = currentNode.next;
    }

    return count;
  }

  search(value) {
    let currentNode = this.head;

    while (currentNode && currentNode.value !== value) {
      currentNode = currentNode.next;
    }

    return Boolean(currentNode);
  }

  remove(value) {
    if (this.isEmpty()) {
      return;
    }

    if (this.head.value === value) {
      this.head = this.head.next;
      return;
    }

    let currentNode = this.head;

    while (currentNode.next) {
      if (currentNode.next.value === value) {
        currentNode.next = currentNode.next.next;
        return;
      }

      currentNode = currentNode.next;
    }
  }

  isEmpty() {
    return this.head === null;
  }
}
